rules/transition_support.bzl

```diff
@@ -32,15 +61,110 @@ def _current_apple_platform(apple_fragment, xcode_config):
         target_os_version = target_os_version,
     )
 
-def _cpu_string(environment_arch, platform_type, settings):
-    """Generates a <platform>_<arch> string for the current target based on the given parameters."""
+# PATCH: end
+
+# PATCH: start - private helper to get the `platform_type` from the rules_ios attributes.
+def _rules_ios_platform_type(settings, attr):
+    platform_type = str(settings["//command_line_option:apple_platform_type"])
+    attr_platform_type = getattr(attr, "platform_type", None)
+    attr_platforms = getattr(attr, "platforms", None)
+    fail_on_apple_rule_transition_platform_mismatches = getattr(attr, "fail_on_apple_rule_transition_platform_mismatches", False)
+    if attr_platform_type and attr_platform_type != platform_type:
+        if fail_on_apple_rule_transition_platform_mismatches:
+            fail("ERROR: {}: attribute platform_type set to {}, but inferred to be {}".format(attr.name, attr_platform_type, platform_type))
+        platform_type = attr_platform_type
+
+    if attr_platforms and platform_type not in attr_platforms:
+        if fail_on_apple_rule_transition_platform_mismatches:
+            fail("ERROR: {}: attribute platforms set to {}, but platform inferred to be {}".format(attr.name, attr_platforms, platform_type))
+        platform_type = attr_platforms.keys()[0]
+
+    if not platform_type:
+        fail("ERROR: platform_type not found")
+
+    return platform_type
+
+# PATCH: end
+
+# PATCH: start - private helper to get the `minimum_os_version` from the rules_ios attributes.
+def _rules_ios_minimum_os_version(platform_type, attr):
```

Comment:
This is essentially the custom stuff `rules_ios` needs, i.e. it uses `platforms` dict to determine this stuff. Longer term we should consider matching rules_apple here...